Add Info to Admins to Set Legal Name when Activating Host

SudharakaP opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/4632

![image](https://user-images.githubusercontent.com/12435965/132056876-1ebf0324-f99e-49d3-bcbd-11fc56f806c2.png)

Betree reviewed (commented):
Can we make it more specific? The only case where this message makes sense is when
- You are your own fiscal host (options 2 & 3)
- You are using an exiting entity (either yourself or an organization), because "Legal name" is already there in the form when you create a new one.
 
For other cases, this message doesn't make much sense.

SudharakaP commented:
> Can we make it more specific? The only case where this message makes sense is when
> 
>     * You are your own fiscal host (options 2 & 3)
> 
>     * You are using an exiting entity (either yourself or an organization), because "Legal name" is already there in the form when you create a new one.
> 
> 
> For other cases, this message doesn't make much sense.

@Betree : So meaning we only show this message for options 2, 3 and 4 right? Because that is the only cases this is relevant. I've done that change so that it shows when the user selects options 2, 3 or 4 (self hosted, own host, or chose an existing fiscal host). Does that look better? 🤔 

![legal-name-info-host](https://user-images.githubusercontent.com/12435965/132377906-4892e588-3d52-4bdc-adbd-374f1ca1d0e5.gif)

Betree commented:
> So meaning we only show this message for options 2, 3 and 4 right?

It's not relevant for action 4, because option 4 is to apply to a fiscal host that you don't control, and therefore you cannot change the name for it.

SudharakaP commented:
> > So meaning we only show this message for options 2, 3 and 4 right?
> 
> It's not relevant for action 4, because option 4 is to apply to a fiscal host that you don't control, and therefore you cannot change the name for it.

Ops, that makes perfect sense. 👍🏽 Corrected it.

Betree commented on components/edit-collective/sections/Host.js:
```diff
@@ -495,6 +496,16 @@ class Host extends React.Component {
+            <MessageBox type="info" fontSize="13px" withIcon mb={4}>
+              <FormattedMessage
+                id="collective.edit.host.legalName.info"
+                defaultMessage="Please set the legal name of the host in the Info section under Settings. This is required if your legal name is different than your display name for tax and accounting purposes."
```
Would be great to have this as a link

![image](https://user-images.githubusercontent.com/1556356/132870365-bb1477b3-341a-40fc-8413-c6bf23aa7e80.png)

SudharakaP replied:
@Betree : I've refactored based on your reviews. Now the info message appears only when the user has already chosen the self hosted options. Here's a screen-cast.

![legal-name-info-host](https://user-images.githubusercontent.com/12435965/132895323-90e9a8e3-e7d0-4f2b-9d48-feca6dbfc7c6.gif)

SudharakaP replied:
Yes, I've done that. 👍🏽

Betree reviewed (commented):
Displaying that at the bottom of the page looks strange to me. It's very to miss, and not displayed at the right time: this info only makes sense when the user already chose to be self-hosted to make sure the host they chose has its legal name properly set.

Here's where I would expect to see it:

- When users pick "Ourselves", on the success page:
![image](https://user-images.githubusercontent.com/1556356/132870763-93a4842f-891f-4ca6-8eeb-f284f2411052.png)

- When users pick "Our organization", there again on the success page:
![image](https://user-images.githubusercontent.com/1556356/132871090-132ead63-8fdd-4a41-9cab-6cbfeba453ee.png)

SudharakaP commented:
> Displaying that at the bottom of the page looks strange to me. It's very to miss, and not displayed at the right time: this info only makes sense when the user already chose to be self-hosted to make sure the host they chose has its legal name properly set.

@Betree : I've refactored based on your reviews. Now the info message appears only when the user has already chosen the self hosted options. Here's a screen-cast.

![legal-name-info-host](https://user-images.githubusercontent.com/12435965/132895323-90e9a8e3-e7d0-4f2b-9d48-feca6dbfc7c6.gif)

Betree commented on components/edit-collective/sections/Host.js:
```diff
@@ -88,6 +88,20 @@ class Host extends React.Component {
+      <MessageBox type="info" fontSize="13px" withIcon>
+        <FormattedMessage
+          id="collective.edit.host.legalName.info"
+          defaultMessage="Please set the legal name of the host in the Info section under {settingsLink}. This is required if your legal name is different than your display name for tax and accounting purposes."
+          values={{
+            settingsLink: <Link href={`/${collective.host?.slug}/edit`}>Settings</Link>,
+          }}
```
The word `Settings` won't be translated here, which will look a bit strange :wink: 

> Veuillez définir le nom légal de l'hôte dans la section info sous settings. Ceci est nécessaire si votre nom légal est différent de votre nom d'affichage pour des raisons fiscales et comptables.

As per https://docs.opencollective.com/help/contributing/development/translations#translate-links-inline, the simplest approach here is `getI18nLink`:
```suggestion
          defaultMessage="Please set the legal name of the host in the Info section under <SettingsLink>your settings</SettingsLink>. This is required if your legal name is different than your display name for tax and accounting purposes."
          values={{
            SettingsLink: getI18nLink({ href: `/${collective.host?.slug}/edit` })
          }}
```

SudharakaP replied:
@Betree : Oh, that's a good point. I didn't think about this one. Have to keep this in mind. 🤔

Betree reviewed (commented):
@SudharakaP You shared the wrong screencast in https://github.com/opencollective/opencollective-frontend/pull/6837#issuecomment-917090800, however, I've tested and the feature looks perfect now. One small feedback about i18n below and we'll be good to merge :rocket:

SudharakaP commented:
> @SudharakaP You shared the wrong screencast in [#6837 (comment)](https://github.com/opencollective/opencollective-frontend/pull/6837#issuecomment-917090800), however, I've tested and the feature looks perfect now. One small feedback about i18n below and we'll be good to merge rocket

@Betree : Sorry about the sreen-cast, somehow I messed up. I've updated the screen-cast with the correct one. I've also updated the `FormattedMessage` as suggested. Hopefully now it looks good. 🙂

Betree reviewed (approved):
looks good
